packages/wallet/src/wallet.ts

```diff
@@ -1,40 +1,172 @@
 import { WalletCore } from '@trustwallet/wallet-core';
 import { HDWallet } from '@trustwallet/wallet-core/dist/src/wallet-core';
 
+// Enum to define mnemonic strength options
+export enum MnemonicStrength {
+    Normal = 128, // 12 words
+    High = 256 // 24 words
+}
+/**
+ * Wallet Data Interface
+ * Represents the data structure for a wallet, including mnemonic, addresses, and next Ed25519 index
+ */
+export interface WalletData {
+    mnemonic?: string;
+    addresses: Array<{
+        path: string;
+        address: string;
+        label: string;
+    }>;
+    nextEd25519Index: number;
+}
 
+/**
+ * Wallet Class
+ * Represents a wallet instance with methods for creating, restoring, and managing addresses
+ */
 export class Wallet {
     private wallet: HDWallet;
     private core: WalletCore;
-    private nextEd25519Index: number; // TODO: save inside JSON
+    private nextEd25519Index: number;
+    private addresses: Array<{ path: string; address: string; label: string }> = [];
 
     private constructor(core: WalletCore, wallet: HDWallet, password: string) {
         this.core = core;
         this.wallet = wallet;
         this.nextEd25519Index = 0;
+        this.addresses = [];
+    }
+
+    /**
+     * Create a new wallet with the specified entropy strength
+     * @param core WalletCore instance
+     * @param strength Mnemonic strength (128 for 12 words, 256 for 24 words)
+     * @param password Password for encryption
+     * @returns A new Wallet instance
+     */
+    static create(
+        core: WalletCore,
+        strength: MnemonicStrength = MnemonicStrength.Normal,
+        password: string
+    ): Wallet {
+        let wallet = core.HDWallet.create(strength, '');
+        return new Wallet(core, wallet, password);
     }
 
-    static create(core: WalletCore, entropy: number, password: string): Wallet {
-        let wallet = core.HDWallet.create(entropy, "");
+    /**
+     * Validate a mnemonic phrase
+     * @param mnemonic The mnemonic phrase to validate
+     * @returns An object with validity status and error message if invalid
+     */
+    static validateMnemonic(mnemonic: string): { isValid: boolean; error?: string } {
```

Comment:
This code looks unnecessary to me. 
If the mnemonic is invalid in any of these cases, the seed won't be generated, and an error (or exception) should be returned naturally.

Think of it this way: ask this object, "What is your main responsibility?"
It would answer, "Managing the wallet, not checking if the seed is valid."

Checking validity should be done somewhere else to keep things organized.